### ISSUE TITLE
Asm( improvements

### ISSUE THREAD

**Issue by commandblockguy** [PeterTillema/ICE#145]
The user should be able to do something like `Asm(21000000)->VAR` to store the value of the hl register after the assembly was executed into an ICE variable.
It would also be useful to be able to add ICE values to the stack prior to running the assembly code by doing something like this: `Asm(DDE5DD21000000DD39DD2709DD070C09DDE1, 69, 420)->VAR`.
This would allow assembly hex codes to interact with the ICE programs that contain them more easily.

**Comment by PeterTillema** [comment]
> 
> 
> `Asm(DDE5DD21000000DD39DD2709DD070C09DDE1, 69, 420)->VAR`.

What does this code do? Push 420 to the stack, then 69, then executes the assembly?

**Comment by commandblockguy** [comment]
Correct, I probably should have specified that. The assembly is just a simple example that would add the two numbers and return the result in hl.

**Comment by PeterTillema** [comment]
Sounds like a good idea, I will see if I have time (and motivation) to implement this somehow. Probably need to change a lot (since now `Asm(` can be used inline), so idk when it's ready.
